### DISCUSSION TITLE
A tool like onnx-modifier for openvino to change model constants?

### DISCUSSION THREAD

**Discussion by cansik** [openvinotoolkit/openvino#18342] [Q&A]
I am currently working with a model that contains a `NonMaxSuppression` operator with fixed score and iou threshold. Does anyone know a tool like [onnx-modifier](https://github.com/ZhangGe6/onnx-modifier) to change openvino models and weights and adjust the constants?

**Comment by Ganesh-403** [comment]
@cansik From what I can tell, there isn't an **OpenVINO equivalent of `onnx-modifier`** that lets you interactively edit an IR model's graph and constants.

However, you do have a couple of options:

1. **Modify the model before conversion (recommended)**

   If you still have the ONNX model, using `onnx-modifier` (or another ONNX graph editing tool) to change the `NonMaxSuppression` thresholds before converting to OpenVINO is probably the simplest workflow.

2. **Modify the OpenVINO graph programmatically**

   OpenVINO exposes the model as a computational graph, and operations such as `NonMaxSuppression` take `iou_threshold` and `score_threshold` as input nodes rather than fixed attributes. You can construct new `Constant` nodes and replace the existing inputs or replace the entire `NonMaxSuppression` node using the graph transformation APIs (e.g. `ov::replace_node`). :contentReference[oaicite:0]{index=0}

For example, the graph rewrite would conceptually look like:

```cpp
auto new_score = ov::op::v0::Constant::create(...);
auto new_iou   = ov::op::v0::Constant::create(...);

// Recreate NMS with the desired constants
// or replace the corresponding constant inputs.
```

I couldn't find a standalone GUI or CLI tool that performs this kind of IR editing automatically, so if your only artifact is the OpenVINO IR, the supported approach appears to be modifying the graph through the OpenVINO Runtime/Transformation API.

One question:

- Do you still have the original **ONNX** model, or only the generated **`.xml` + `.bin`** OpenVINO IR?

That determines whether it's easier to modify the graph before conversion or perform a graph rewrite on the IR itself.

If this solves your problem, feel free to mark it as the accepted answer so others can find it easily.
